metricbeat-tests/runner_test.go

```diff
@@ -68,23 +84,11 @@ func (mts *MetricbeatTestSuite) CleanUp() error {
 	return err
 }
 
-// As we are using an index per scenario outline, with an index name formed by metricbeat-version1-module-version2,
-// and because of the ILM is configured on metricbeat side, then we can use an asterisk for the index name:
-// each scenario outline will be namespaced, so no collitions between different test cases should appear
-func (mts *MetricbeatTestSuite) getIndexName() string {
-	mVersion := strings.ReplaceAll(mts.Version, "-SNAPSHOT", "")
-
-	index := fmt.Sprintf("metricbeat-%s-%s-%s", mVersion, mts.ServiceName, mts.ServiceVersion)
-
-	index += "-test*"
-
-	return index
-}
-
 func (mts *MetricbeatTestSuite) installedAndConfiguredForModule(version string, serviceType string) error {
 	serviceType = strings.ToLower(serviceType)
 
-	mts.Version = version
+	// at this point we have everything to define the index name
+	mts.setIndexName(version)
```

Comment:
It looks a bit counterintuitive to set also the version in `setIndexName`. What do you think about explicitly set the version as before with `mts.Version = version`, and instead of having a set method for the index name, just have a `IndexName() string` method that generates and stores the index name for following calls.